crates/guest/src/guest.rs

```diff
@@ -80,9 +83,12 @@ where
 
     // Deserialize the host bytes into the output type.
     let bytes: Vec<u8> = crate::allocation::consume_bytes(output_guest_ptr);
-    match holochain_serialized_bytes::decode(&bytes) {
-        Ok(output) => Ok(output),
-        Err(_) => Err(WasmError::Deserialize(bytes)),
+    match holochain_serialized_bytes::decode::<Vec<u8>, Result<O, WasmError>>(&bytes) {
+        Ok(output) => Ok(output?),
```

Comment:
the addition of `?` in this `Ok` here handles the 'inner error', i.e. first we handle the guest then we handle the host, and the caller of `host_call` sees a unified `Result`, where the `WasmError` enum itself contains host/guest source information, if we didn't do this then the consumers of the HDK will need to be doing `??` everywhere or similar